Scripts/Runtime/Logging/.CSProject/UnityLogListener.cs

```diff
@@ -267,47 +297,48 @@ private void ProcessPendingLogs()
+        private void SendToLogger(UnityEngine.Object context, LogLevel logLevel, string message, StackTrace stackTrace = null)
         {
-            // Skip 4 frames to get to the original caller
-            // 0 - Here
-            // 1 - LogException or LogFormat
-            // 2 - UnityEngine.Logger.Log(+Warning, +Error, ...), Assert
-            // 3 - UnityEngine.Debug.Log(+Warning, +Error, ...), Assert
-            // 4 - Caller of Debug.Log(+Warning, +Error, ...), Assert
-            (StackFrame callerFrame, MethodBase callerMethod) = ResolveCaller(4);
+            (StackFrame callerFrame, MethodBase callerMethod) = ResolveCaller(stackTrace);
+
+            string callerPath = callerFrame?.GetFileName() ?? callerMethod?.ReflectedType.Name;
 
             Logger logger = context != null ? Log.GetLogger(context) : Log.GetStaticLogger(ResolveContextFromMethod(callerMethod));
-            logger.AtLevel(logLevel, message, callerFrame.GetFileName(), callerMethod?.Name, callerFrame.GetFileLineNumber());
+            logger.AtLevel(logLevel, message, callerPath, callerMethod?.Name, callerFrame?.GetFileLineNumber() ?? 0);
         }
 
-        private (StackFrame callerFrame, MethodBase callerMethod) ResolveCaller(int skipFrames)
+        private (StackFrame callerFrame, MethodBase callerMethod) ResolveCaller(StackTrace stackTrace = null)
         {
-            StackFrame callerFrame;
-            MethodBase callerMethod;
-            // Walk up the stack until we're at a caller that isn't Unity's proxy or a method that we
-            // want to skip (Ex: Logger objects).
-            do
+            // If no explicit stack trace is given, skip 3 frames (ResolveCaller, SendToLogger, and LogFormat/LogException)
+            int frameIndex = (stackTrace == null ? 3 : 0);
+            StackFrame frame;
+
+            while ((frame = GetNextFrame()) != null)
             {
-                skipFrames++;
-                callerFrame = new StackFrame(skipFrames, true);
-                callerMethod = callerFrame.GetMethod();
-            } while (
-                callerMethod != null
-                // Skip methods with the skip attribute
-                && (
-                    callerMethod.GetCustomAttribute<ExcludeAttribute>(true) != null
-
-                    // Check to see if this call went through Unity's new stubbed out logging class that
-                    // currently just proxies the existing logging system.
-                    // This was found in the v0.17 of the Unity.Entities package
-                    || callerMethod.ReflectedType.FullName == "Unity.Debug"
-                    )
-                );
-
-            return (callerFrame, callerMethod);
+                MethodBase method = frame.GetMethod();
+
+                ExcludeAttribute excludeAttribute = method.GetCustomAttribute<ExcludeAttribute>(inherit: true);
+                if (excludeAttribute != null) continue;
```

Comment:
Minor but we like to keep the squigglies. 
```suggestion
                if (excludeAttribute != null) 
                {
                    continue;
                }
```

Reply:
yes...we...
I joke, it is the consistent style of the library.